src/screens/SearchScreen.js

```diff
@@ -37,42 +40,43 @@ const SearchScreen = () => {
           term={term}
           onTermChange={(newTerm) => setTerm(newTerm)}
           onTermSubmit={() => searchApi(term, city)}
+          onEditStart={() => setTerm('')}
         />
         <CityButton 
-          toOpenModal={() => setModalVisible(true)}
+          onPress={() => setModalVisible(true)}
         />
       </View>
       {ShowErrorMsg(errorMsg)}
       <ResultList
         results = {filteredRes[0]}
         title="An Easy Rate"
         ready={contentReady}
+        onPress={shopId => navigation.navigate('MoreInfo', {id: shopId})}
       />
       <ResultList
         results = {filteredRes[1]}
         title="Average Price"
         ready={contentReady}
+        onPress={shopId => navigation.navigate('MoreInfo', {id: shopId})}
       />
-      <ResultList 
+      <ResultList
         results = {filteredRes[2]}
         title="Big Spender"
         ready={contentReady}
+        onPress={shopId => navigation.navigate('MoreInfo', {id: shopId})}
       />
       <Modal
         animationType="slide"
         visible={modalVisible}
       >
         <ModalCloseButton 
-          toCloseModal={() => setModalVisible(false)}
+          onPress={() => setModalVisible(false)}
         />
```

Comment:
Change the names of simple callback functions to 'onPress', to make things more simple.